src/components/Modal/index.tsx

```diff
@@ -20,12 +20,14 @@ const Modal: React.FC<Props> = props => {
       }
     };
 
-    document.addEventListener('mousedown', handleClickOutside);
+    open
+      ? document.addEventListener('mousedown', handleClickOutside)
+      : document.removeEventListener('mousedown', handleClickOutside);
```

Comment:
@viniciusaa instead of having a ternary, you can simply check from the `open` property and add the event listener.

Reply:
I couldn't understand @paulohfev . Do you mean using this:
```
open && document.addEventListener('mousedown', handleClickOutside)
!open && document.removeEventListener('mousedown', handleClickOutside);
```

Reply:
@viniciusaa I mean this: 
`if (open) { document.addEventListener('mousedown', handleClickOutside) }`

Reply:
This wouldn't do. I would need to do this:
```
if (open) { document.addEventListener('mousedown', handleClickOutside) }
else {
   document.removeEventListener('mousedown', handleClickOutside);
}
```
And this is basically the same code that I'm currently using.

Reply:
Note that `addEventListener` and `removeEventListener` are different things.